platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.formatting.TextTable;
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.common.merkle.utility.MerkleTreeVisualizer;
+import com.swirlds.platform.consensus.ConsensusSnapshot;
+import java.util.List;
+
+public class MerkleStateUtils {
+    /**
+     * Generate a string that describes this state.
+     *
+     * @param hashDepth the depth of the tree to visit and print
+     * @param platformState current platform state
+     * @param state current root node state
+     *
+     */
+    public static String createInfoString(int hashDepth, PlatformState platformState, Hash rootHash, MerkleNode state) {
```

Comment:
if the code in this module is not exempt from using the `final` keyword or `@NonNull`, maybe here is a good place to start

Reply:
I added `final` modifiers and `@NonNull` annotation as you recommended.